core/templates/pages/contributor-dashboard-page/badge/badge.component.ts

```diff
@@ -0,0 +1,71 @@
+// Copyright 2022 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for the badge.
+ */
+
+import { Component, Input } from '@angular/core';
+import { AppConstants } from 'app.constants';
+
+@Component({
+  selector: 'badge',
+  templateUrl: './badge.component.html',
+  styleUrls: []
+})
+export class BadgeComponent {
+  @Input() contributionType!: string;
+  @Input() contributionSubType!: string;
+  @Input() contributionCount!: number;
+  @Input() language!: string | null;
+  @Input() isUnlocked!: boolean;
+  contributionSubTypeText!: string;
+
+  fontSize: string = '13px';
+  lineHeight: string = '20px';
+
+  constructor() {}
+
+  ngOnInit(): void {
+    if (
+      this.contributionSubType === AppConstants
+        .CONTRIBUTION_STATS_SUBTYPE_SUBMISSION) {
+      this.contributionSubTypeText = 'Submission';
+    } else if (this.contributionSubType === AppConstants
+      .CONTRIBUTION_STATS_SUBTYPE_REVIEW) {
+      this.contributionSubTypeText = 'Review';
+    } else if (
+      this.contributionSubType === AppConstants
+        .CONTRIBUTION_STATS_SUBTYPE_CORRECTION
+    ) {
+      this.contributionSubTypeText = 'Correction';
+    } else {
```

Comment:
You can implement mapping from the AppConstants.XXX → 'yyy' and then just do `mapping[XXX]` here to remove th ifs.

Reply:
Done